File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_141_file_name_check.lua
-- Create a function which takes a string representing a file's name, and returns
-- 'Yes' if the the file's name is valid, and returns 'No' otherwise.
-- A file's name is considered to be valid if and only if all the following conditions 
-- are met:
-- - There should not be more than three digits ('0'-'9') in the file's name.
-- - The file's name contains exactly one dot '.'
-- - The substring before the dot should not be empty, and it starts with a letter from 
-- the latin alphapet ('a'-'z' and 'A'-'Z').
-- - The substring after the dot should be one of these: ['txt', 'exe', 'dll']
-- Examples:
-- file_name_check("example.txt") # => 'Yes'
-- file_name_check("1example.dll") # => 'No' (the name should start with a latin alphapet letter)
local function file_name_check(file_name)
  return string.match("^[0-9][0-9][0-9][0-9]" .. file_name .. ".*$", "^[^.]*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    string.match("^[^.]*$", "^[a-z][a-z][a-z]" .. file_name .. ".*$") ~= nil and
    

lu = require('luaunit')

function test_humaneval()
local candidate = file_name_check
    lu.assertEquals(candidate('example.txt'), 'Yes')
    lu.assertEquals(candidate('1example.dll'), 'No')
    lu.assertEquals(candidate('s1sdf3.asd'), 'No')
    lu.assertEquals(candidate('K.dll'), 'Yes')
    lu.assertEquals(candidate('MY16FILE3.exe'), 'Yes')
    lu.assertEquals(candidate('His12FILE94.exe'), 'No')
    lu.assertEquals(candidate('_Y.txt'), 'No')
    lu.assertEquals(candidate('?aREYA.exe'), 'No')
    lu.assertEquals(candidate('/this_is_valid.dll'), 'No')
    lu.assertEquals(candidate('this_is_valid.wow'), 'No')
    lu.assertEquals(candidate('this_is_valid.txt'), 'Yes')
    lu.assertEquals(candidate('this_is_valid.txtexe'), 'No')
    lu.assertEquals(candidate('#this2_i4s_5valid.ten'), 'No')
    lu.assertEquals(candidate('@this1_is6_valid.exe'), 'No')
    lu.assertEquals(candidate('this_is_12valid.6exe4.txt'), 'No')
    lu.assertEquals(candidate('all.exe.txt'), 'No')
    lu.assertEquals(candidate('I563_No.exe'), 'Yes')
    lu.assertEquals(candidate('Is3youfault.txt'), 'Yes')
    lu.assertEquals(candidate('no_one#knows.dll'), 'Yes')
    lu.assertEquals(candidate('1I563_Yes3.exe'), 'No')
    lu.assertEquals(candidate('I563_Yes3.txtt'), 'No')
    lu.assertEquals(candidate('final..txt'), 'No')
    lu.assertEquals(candidate('final132'), 'No')
    lu.assertEquals(candidate('_f4indsartal132.'), 'No')
    lu.assertEquals(candidate('.txt'), 'No')
    lu.assertEquals(candidate('s.'), 'No')
end

os.exit(lu.LuaUnit.run())

File: datasets/php-keep-code_davinci_001_temp_0.2/HumanEval_141_file_name_check.php
<?php
// Create a function which takes a string representing a file's name, and returns
// 'Yes' if the the file's name is valid, and returns 'No' otherwise.
// A file's name is considered to be valid if and only if all the following conditions 
// are met:
// - There should not be more than three digits ('0'-'9') in the file's name.
// - The file's name contains exactly one dot '.'
// - The substring before the dot should not be empty, and it starts with a letter from 
// the latin alphapet ('a'-'z' and 'A'-'Z').
// - The substring after the dot should be one of these: ['txt', 'exe', 'dll']
// Examples:
// file_name_check("example.txt") # => 'Yes'
// file_name_check("1example.dll") # => 'No' (the name should start with a latin alphapet letter)
function file_name_check($file_name) {
    $file_name = strtolower($file_name);
    $file_name = explode('.', $file_name);
    $file_name = array_filter($file_name);
    $file_name = array_values($file_name);
    if (count($file_name) > 2) {
        return 'No';
    }
    if (count($file_name) == 1) {
        return 'No';
    }
    if (count($file_name) == 2) {
        if (preg_match('/[a-zA-Z]/', $file_name[0]) == 0) {
            return 'No';
        }
        if (preg_match('/[a-zA-Z]/', $file_name[1]) == 0) {
            return 'No';
        }
        if (preg_match('/[0-9]/', $file_name[0]) == 1) {
            return 'No';
        }
        if (preg_match('/[0-9]/', $file_name[1]) == 1) {
            return 'No';
        }
        if (preg_match('/[a-zA-Z]/', $file_name[0]) == 1) {
            if (preg_match('/[a-zA-Z]/', $file_name[1]) == 1) {
                if (preg_match('/[0-9]/', $file_name[0]) == 0) {
                    if (preg_match('/[0-9]/', $file_name[1]) == 0) {
                        if (in_array($file_name[1], ['txt', 'exe', 'dll'])) {
                            return 'Yes';
                        }
                        return 'No';
                    }
                }
            }
        }
    }
}

$file_name = 'example.txt';
$file_name = '1example.dll';
$file_name = 'example.exe';
$file_name = 'example.dll';
$

function candidate(...$args) {
    return file_name_check(...$args);
}

function test(): void {
    if (candidate("example.txt") !== "Yes") { throw new Exception("Test failed!"); }
    if (candidate("1example.dll") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("s1sdf3.asd") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("K.dll") !== "Yes") { throw new Exception("Test failed!"); }
    if (candidate("MY16FILE3.exe") !== "Yes") { throw new Exception("Test failed!"); }
    if (candidate("His12FILE94.exe") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("_Y.txt") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("?aREYA.exe") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("/this_is_valid.dll") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("this_is_valid.wow") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("this_is_valid.txt") !== "Yes") { throw new Exception("Test failed!"); }
    if (candidate("this_is_valid.txtexe") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("#this2_i4s_5valid.ten") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("@this1_is6_valid.exe") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("this_is_12valid.6exe4.txt") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("all.exe.txt") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("I563_No.exe") !== "Yes") { throw new Exception("Test failed!"); }
    if (candidate("Is3youfault.txt") !== "Yes") { throw new Exception("Test failed!"); }
    if (candidate("no_one#knows.dll") !== "Yes") { throw new Exception("Test failed!"); }
    if (candidate("1I563_Yes3.exe") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("I563_Yes3.txtt") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("final..txt") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("final132") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("_f4indsartal132.") !== "No") { throw new Exception("Test failed!"); }
    if (candidate(".txt") !== "No") { throw new Exception("Test failed!"); }
    if (candidate("s.") !== "No") { throw new Exception("Test failed!"); }
}

test();